src/writeckpt.cpp

```diff
@@ -228,7 +234,8 @@ void mtcp_writememoryareas(int fd)
       area.name[0] = '\0';
     } else if (Util::isNscdArea(area)) {
       /* Special Case Handling: nscd is enabled*/
-      area.prot = PROT_READ | PROT_WRITE | MTCP_PROT_ZERO_PAGE;
+      area.prot = PROT_READ | PROT_WRITE;
+      area.properties = DMTCP_ZERO_PAGE;
```

Comment:
There's a minor inconsistency here (and throughout this patch).

This version assumes that area.properties is equal to DMTCP_ZERO_PAGE or DMTCP_SKIP_WRITING_TEXT_AREA or 0.  Yet, you defined the macros as 0x0001 and 0x0002, which implies that these are bit masks, and that you can "or" them together to have both properties.  If they are bit properties, then `|=` is the right operator.  Otherwise, the macros should be set to `1` and `2`, rather than `0x0001` and `0x0002`.

In the future, you might switch between bit masks and enumerated types.  But for now, choose one or the other.